pyresample/test/test_geometry.py

```diff
@@ -1544,6 +1644,18 @@ def test_create_area_def(self):
         self.assertEqual(area_def.shape, (101, 90))
         # Checks every area definition made
         for area_def in area_list:
+            if 'EPSG' in area_def.proj_dict or 'init' in area_def.proj_dict:
+                # Use formal definition of EPSG projections to make them comparable to the base definition
+                proj_def = pyproj.Proj(area_def.proj_str).definition_string().strip()
+                area_def = area_def.copy(projection=proj_def)
+
+                # Remove extra attributes from the formal definition
+                if 'R' in area_def.proj_dict:
+                    # pyproj < 2
+                    area_def.proj_dict['a'] = area_def.proj_dict.pop('R')
+                for key in ['x_0', 'y_0', 'no_defs', 'b', 'init']:
+                    area_def.proj_dict.pop(key, None)
+
```

Comment:
What is the above actually testing? Is this showing that area definitions with EPSG codes aren't comparable out of the box?

Reply:
The generated area defs are compared to `base_def` whos `proj_dict` looks like `{'proj': 'laea', 'lat_0': -90, ...}`. However, the EPSG area defs have `proj_dict={'EPSG': 1234}` so that they are not directly comparable to `base_def`. The workaround is to use their `definition_string`. But that contains some additional (redundant) parameters like `x_0=0, y_0=0`. The ellipsoid parameters in the definition string are also different depending on the proj version (`a` and `b` in pyproj >= 2 vs `R` in pyproj < 2)